src/main/java/io/leonis/subra/ipc/serialization/gson/GoalSerializer.java

```diff
@@ -19,10 +19,10 @@ public JsonElement serialize(
     final JsonObject jsonObject = new JsonObject();
     jsonObject.addProperty("x", src.getX());
     jsonObject.addProperty("y", src.getY());
-    jsonObject.addProperty("width", src.getWidth());
-    jsonObject.addProperty("depth", src.getDepth());
-    jsonObject.add("cardinalDirection", context.serialize(src.getCardinalDirection()));
-    jsonObject.add("teamColor", context.serialize(src.getTeamColor()));
+    jsonObject.addProperty("width", src.getGoalDimension().getWidth());
+    jsonObject.addProperty("depth", src.getGoalDimension().getDepth());
+    jsonObject.add("fieldHalf", context.serialize(src.getFieldHalf()));
+    jsonObject.add("teamIdentity", context.serialize(src.getTeamIdentity()));
```

Comment:
does this serialization happen correctly? dont we need a handler for this? (can't check properly from this phone ui)

Reply:
Good question, it should be fine in our case, since `TeamColor` is an enum, which overrides `#toString`. 

```java
  public static void main(String[] args) {
    TeamIdentity tid = TeamColor.BLUE;
    System.out.println(tid);
    System.out.println(TeamColor.BLUE);
  }
```
Returns "BLUE" in both cases, as expected, so we can remove `context.serialize(` in front.